test_conformance/compiler/test_build_helpers.cpp

```diff
@@ -240,18 +246,22 @@ int test_load_discreet_length_source(cl_device_id deviceID, cl_context context,
 
     int i;
 
+    constexpr int num_lines = ARRAY_SIZE(sample_kernel_code_bad_multi_line);
 
     /* Preprocess: calc the length of each source file line */
-    for( i = 0; i < 7; i++ )
+    size_t line_lengths[num_lines];
+    for (i = 0; i < num_lines; i++)
     {
-        sample_bad_multi_line_lengths[ i ] = strlen( sample_kernel_code_bad_multi_line[ i ] );
+        line_lengths[i] = strlen(sample_kernel_code_bad_multi_line[i]);
     }
 
     /* Now force the length of the third line to skip the actual error */
-    sample_bad_multi_line_lengths[2] -= strlen("thisisanerror");
+    line_lengths[2] -= strlen("thisisanerror");
```

Comment:
We could statically assert that `line_lengths` is at least 3 elements.

Reply:
Done.